homeassistant/components/transmission/__init__.py

```diff
@@ -181,19 +178,31 @@ def add_torrent(service):
                 )
 
         self.hass.services.async_register(
-            DOMAIN, SERVICE_ADD_TORRENT, add_torrent, schema=SERVICE_ADD_TORRENT_SCHEMA
+            DOMAIN, self.service_name, add_torrent, schema=SERVICE_ADD_TORRENT_SCHEMA
         )
 
         self.config_entry.add_update_listener(self.async_options_updated)
 
         return True
 
+    def add_options(self):
+        """Add options for entry."""
+        if not self.config_entry.options:
+            scan_interval = self.config_entry.data.pop(
+                CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+            )
+            options = {CONF_SCAN_INTERVAL: scan_interval}
+
+            self.hass.config_entries.async_update_entry(
+                self.config_entry, options=options
+            )
+
     def set_scan_interval(self, scan_interval):
         """Update scan interval."""
 
-        def refresh(event_time):
+        async def refresh(event_time):
```

Comment:
This should be reverted. The `TransmissionData` instance methods are not async safe.